shared/src/commonTest/kotlin/vn/luongvo/kmm/survey/data/repository/AuthRepositoryTest.kt

```diff
@@ -0,0 +1,53 @@
+package vn.luongvo.kmm.survey.data.repository
+
+import io.kotest.matchers.shouldBe
+import io.mockative.*
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.test.runTest
+import vn.luongvo.kmm.survey.data.remote.datasource.AuthRemoteDataSource
+import vn.luongvo.kmm.survey.data.remote.model.response.toToken
+import vn.luongvo.kmm.survey.domain.repository.AuthRepository
+import vn.luongvo.kmm.survey.test.tokenResponse
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+
+@ExperimentalCoroutinesApi
+class AuthRepositoryTest {
+
+    @Mock
+    private val mockAuthRemoteDataSource = mock(AuthRemoteDataSource::class)
+
+    private lateinit var repository: AuthRepository
+
+    @BeforeTest
+    fun setUp() {
+        repository = AuthRepositoryImpl(mockAuthRemoteDataSource)
+    }
+
+    @Test
+    fun `when calling logIn successfully - it returns token`() = runTest {
+        given(mockAuthRemoteDataSource)
+            .suspendFunction(mockAuthRemoteDataSource::logIn)
+            .whenInvokedWith(any())
+            .thenReturn(tokenResponse)
+
+        repository.logIn("email", "password").collect {
+            it shouldBe tokenResponse.toToken()
+        }
+    }
+
+    @Test
+    fun `when calling logIn fails - it throws error`() = runTest {
+        val mockThrowable = Throwable()
+        given(mockAuthRemoteDataSource)
+            .suspendFunction(mockAuthRemoteDataSource::logIn)
+            .whenInvokedWith(any())
+            .thenThrow(mockThrowable)
+
+        repository.logIn("email", "password").catch {
```

Comment:
Su pointed me to this [tool](https://github.com/cashapp/turbine), might be useful for you 😄

Reply:
@blyscuit I intended to use it in ViewModel test in `android` module. But yeah, it makes sense to use it in `shared` module with shorter implementation & more utilities. 👍

Reply:
Fixed in https://github.com/luongvo/kmm-survey/pull/50/commits/06e30d07cba0d5b8a026204ff962e9ca01eccc46